### DISCUSSION TITLE
MIDI Callback - How to access to the Metadata table ?

### DISCUSSION THREAD

**Discussion by Badsheep** [Hammerspoon/hammerspoon#3381] [Q&A] [answered]
Hello,

I do a MIDI Callback like this : 

```
function ReceivedMidi(object, deviceName, commandType, description, metadata)

  print("object: " .. tostring(object))
  print("deviceName: " .. deviceName)
  print("commandType: " .. commandType)
  print("description: " .. description)
  print("metadata: " .. hs.inspect(metadata))

    hs.alert.show(metadata[2])
end

fromMax:callback(ReceivedMidi)
```

Everything seems to work fine, the console displays : 

2023-02-13 14:59:32: object: hs.midi: from Max 1 (0x606000069958)
2023-02-13 14:59:32: deviceName: from Max 1
2023-02-13 14:59:32: commandType: controlChange
2023-02-13 14:59:32: description: <MIKMIDIControlChangeCommand: 0x604000b7eed0> time: 14:59:32.566 command: 191 channel 0 control number: 4 value: 127 14-bit? 0 
	data: {length = 3, bytes = 0xb0047f}
2023-02-13 14:59:32: metadata: {
  channel = 0,
  controllerNumber = 4,
  controllerValue = 127,
  data = "b0047f",
  fourteenBitCommand = false,
  fourteenBitValue = 16256,
  isVirtual = true,
  timestamp = "14:59:32.566"
}

But still I don't know how to access to the values stored in "metadata"...

I tried 
`hs.alert.show(metadata[2])`
and
`hs.alert.show(metadata[controllerNumber])`

I always get nil.

But when the hs.inspect from above works. ??
I don't understand why I can't get datas from this table...
Thanks for your help !

**Comment by Badsheep** [comment, marked as answer]
I found the answer in a script on the web... The correct way to do it is : 

`hs.alert.show(metadata["controllerNumber"])
`

Sorry this thread was useless, problem solved.

**Reply by latenitefilms** [reply]
You can also do `metadata.controllerNumber`
